ampersand-calendar.js

```diff
@@ -0,0 +1,290 @@
+;(function() {
+  var d3 = require('d3');
+  var _ = require('lodash');
+  var moment = require('moment');
+
+  var AmpersandState = require('ampersand-state');
+  var AmpersandView = require('ampersand-view');
+
+  var CalendarState = AmpersandState.extend({
+    session: {
+      startDate: [ 'object', false, function() { return null; } ],
+      endDate: [ 'object', false, function() { return null; } ],
+      current: [ 'object', false, function() { return moment(); } ],
+      currentDates: 'array'
+    },
+    initialize: function() {
+      this.currentDates = this.createDateArray();
+    },
+    selectPreviousMonth: function() {
+      this.current.subtract(1, 'M');
+      this.currentDates = this.createDateArray();
+    },
+    selectNextMonth: function() {
+      if (this.current.year() < moment().year() || (this.current.year() === moment().year() && this.current.month() < moment().month())) {
```

Comment:
we'll want a TODO somewhere to update this to allow future dates as well.  An issue on github may be a good way to track that TODO.  I'm thinking we'd likely want to specify an `earliestDate` and `latestDate` that could be null (unlimited)